lib/plugins/aws/provider/awsProvider.js

```diff
@@ -187,6 +187,13 @@ class AwsProvider {
     impl.addEnvironmentProfile(result, `AWS_${stageUpper}`);
 
     result.region = this.getRegion();
+
+    if (this.serverless.service.provider.deploymentBucketObject &&
+        this.serverless.service.provider.deploymentBucketObject.serverSideEncryption &&
+        this.serverless.service.provider.deploymentBucketObject.serverSideEncryption ===
+         'aws:kms') {
+      result.signatureVersion = 'v4';
+    }
```

Comment:
updating the aws-sdk version did not seem to effect the sigv4 signing at all. so I'm leaving the referenced code in place